clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -99,12 +99,8 @@ public void process(ApplicationEvent event) {
                 process((ListOffsetsEvent) event);
                 return;
 
-            case RESET_POSITIONS:
-                process((ResetPositionsEvent) event);
-                return;
-
-            case VALIDATE_POSITIONS:
-                process((ValidatePositionsEvent) event);
+            case UPDATE_FETCH_POSITIONS:
+                process((CheckPositionsAndMaybeUpdate) event);
```

Comment:
This is all so nice, but then the new event type doesn't match the class name, and that convention has been followed so far. I suggest `CHECK_AND_UPDATE_POSITIONS` and `CheckAndUpdatePositionsEvent` as not too wordy and relatively accurate.

Reply:
totally agree, that definitely escaped my consistency check :)